src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php

```diff
@@ -41,35 +41,37 @@ public function register_routes() {
 		// List themes global styles.
 		register_rest_route(
 			$this->namespace,
-			'/' . $this->rest_base . '/themes/(?P<stylesheet>[^.\/]+(?:\/[^.\/]+)?)',
+			'/' . $this->rest_base . '/themes/(?P<stylesheet>[\/\s%\w\.\(\)\[\]\@_\-]+)',
 			array(
 				array(
 					'methods'             => WP_REST_Server::READABLE,
 					'callback'            => array( $this, 'get_theme_item' ),
 					'permission_callback' => array( $this, 'get_theme_item_permissions_check' ),
 					'args'                => array(
 						'stylesheet' => array(
-							'description' => __( 'The theme identifier' ),
-							'type'        => 'string',
+							'description'       => __( 'The theme identifier' ),
+							'type'              => 'string',
+							'sanitize_callback' => array( $this, '_sanitize_global_styles_callback' ),
 						),
 					),
 				),
 			)
 		);
 
-		// Lists/updates a single gloval style variation based on the given id.
+		// Lists/updates a single global style variation based on the given id.
 		register_rest_route(
 			$this->namespace,
-			'/' . $this->rest_base . '/(?P<id>[\/\w-]+)',
+			'/' . $this->rest_base . '/(?P<id>[\/\s%\w\.\(\)\[\]\@_\-]+)',
 			array(
 				array(
 					'methods'             => WP_REST_Server::READABLE,
 					'callback'            => array( $this, 'get_item' ),
 					'permission_callback' => array( $this, 'get_item_permissions_check' ),
 					'args'                => array(
 						'id' => array(
-							'description' => __( 'The id of a template' ),
-							'type'        => 'string',
+							'description'       => __( 'The id of a template' ),
+							'type'              => 'string',
+							'sanitize_callback' => array( $this, '_sanitize_global_styles_callback' ),
```

Comment:
Do we need another method for this? Would this work?
```suggestion
							'sanitize_callback' => 'urldecode',
```

Reply:
I did that first but the callback receives 3 args whereas `urldecode()` only accepts 1. Arguments passed are the ID or stylesheet, request object, and the arg name. Tests fail and PHP Warning is thrown on the site.

```
Warning: urldecode() expects exactly 1 parameter, 3 given
```